src/lib.rs

```diff
@@ -173,7 +172,7 @@ impl Context {
             if fast {
                 flags = flags | randomx_flags_RANDOMX_FLAG_FULL_MEM;
                 dataset = randomx_alloc_dataset(flags);
-                let num_threads = num_cpus::get();
+                let num_threads = thread::available_parallelism().expect("Failed to determine available parallelism").get();
```

Comment:
Maybe `.unwrap_or(1)` instead of `expect()`?

Reply:
Nah, an error is better, otherwise the user may not know not all of his CPU is being consumed.